tls/s2n_internal.h

```diff
@@ -47,3 +47,9 @@ struct s2n_connection;
  */
 S2N_PRIVATE_API
 extern int s2n_connection_get_config(struct s2n_connection *conn, struct s2n_config **config);
+
+/*
+ * Poll the async client_hello callback to make progress.
+ */
+S2N_PRIVATE_API
+extern int s2n_connection_client_hello_cb_poll(struct s2n_connection *conn);
```

Comment:
This doesn't actually poll; it enables polling. Maybe call it `s2n_connection_client_hello_cb_enable_poll` instead?

Reply:
right: one would have to call negotiate multiple times. Ill add that to the comments also